GitCommands/Git/GitModule.cs

```diff
@@ -1219,7 +1219,14 @@ public string GetSubmoduleFullPath(string? localPath)
 
         public GitModule GetSubmodule(string? localPath)
         {
-            return new GitModule(GetSubmoduleFullPath(localPath));
+            try
+            {
+                return new GitModule(GetSubmoduleFullPath(localPath));
+            }
+            catch (GitConfigurationException)
+            {
+                return new GitModule(null);
+            }
```

Comment:
This function `GetSubmodule` is used in many other functions.
Converting an exception into `GitModule(null)` might cause other less clear exceptions.

Rather handle the exception in `SubmoduleHelpers.ParseSubmodulePatchStatus` and return the error in an `GitSubmoduleStatus` instance. (If I have not identified the correct function used by FormCommit, please do something similar in the correct function.)